infra/blueprint-test/README.md

```diff
@@ -0,0 +1,408 @@
+The entire integration test explained below can be found [here](https://github.com/terraform-google-modules/terraform-google-sql-db/blob/master/test/integration/mysql-public/mysql_public_test.go).
+
+The first step in writing the test is to wire it up with the required packages and methods signatures that the test framework expects as follows:
+1. Cd in the `test/integration/mysql-public` directory or create it if it's not present already in the blueprint and the cd into it.
+2. Create file in `mysql_public_test.go` with the following content: 
+
+As a good practice use this convention to name your test files: <example_name>_test.go
+     
+```go
+// define test package name
+package mysql_public
+ 
+import (
+  "fmt"
+  "testing"
+ 
+  // import the blueprints test framework modules for testing and assertions
+  "github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/gcloud"   
+  "github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test/pkg/tft"
+  "github.com/stretchr/testify/assert"
+)
+ 
+// name the function as Test*
+func TestMySqlPublicModule(t *testing.T) {
+   ...
+   // initialize Terraform test from the blueprint test framework
+   mySqlT := tft.NewTFBlueprintTest(t)
+   // define and write a custom verifier for this test case call the default verify for confirming no additional changes
+   mySqlT.DefineVerify(func(assert *assert.Assertions) {
+       mySqlT.DefaultVerify(assert)    
+       // custom logic for the test continues below
+       ...   
+   })
+   // call the test function to execute the integration test
+   mySqlT.Test()
+}
+```
+
+The next step in the process is to write the logic for assertions.
+
+1. In most cases, you will be asserting against values retrieved from the GCP environment. This can be done by using the (gcloud)[https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/blob/master/infra/blueprint-test/pkg/gcloud/gcloud.go] helper in our test framework, which executes gcloud commands and stores their JSON outputs as. The gcloud helper can be initialized as follows:
+
+```go
+// The tft struct can be used to pull output variables of the TF module being invoked by this test
+op := gcloud.Run(t, fmt.Sprintf("sql instances describe %s --project %s", mySqlT.GetStringOutput("name"), mySqlT.GetStringOutput("project_id")))
+```
+
+2. Once you have retrieved values from GCP, use the (assert)[https://pkg.go.dev/github.com/stretchr/testify/assert] package to perform custom validations with respect to the resources provisioned. Here are some common assertions that can be useful in most test scenarios.
+
+   1. Contains
+
+    ```go
+    // assert values that are contained in the expected output
+    assert.Contains(op.Get("gceZone").String(), region, "GCE region is valid")
```

Comment:
for `op.Get("gceZone").String()` shouldn't it be an assert.Equals. Contains would be for a slice?

Reply:
It can assert part of a string, list or map. I just picked gceZone as an example since it would always contain the regional sub-string.

Reply:
oh I see, sgtm